src/main/java/org/terasology/alterationEffects/ComponentBasedAlterationEffect.java

```diff
@@ -0,0 +1,190 @@
+ * and expiring the effect after some duration.
+ * <p>
+ * Identifiers for common alteration effects can be found in {@link AlterationEffects}.
+ * <p>
+ * This skeleton implementation applies an alteration affect as follows:
+ * <ol>
+ *     <li><strong>upsert</strong> the backing component and send {@link OnEffectModifyEvent} to notify systems</li>
+ *     <li><strong>update</strong> the backing component if the effect was modified</li>
+ *     <li>schedule effect expiration or <strong>remove</strong> the backing component</li>
+ * </ol>
+ * <p>
+ * An effect based on this class must be accompanied by the corresponding <emph>backing component class</emph>. In
+ * addition, a <emph>backing system</emph> (or corresponding system) should be implemented to react on the effect
+ * component and handle subsequent {@link OnEffectModifyEvent}s.
+ *
+ * <strong>Note:</strong> I believe this to be only a temporary solution for simplifying alteration effects.
+ *
+ * @param <C> the component class this effect is backed by
+ */
+public abstract class ComponentBasedAlterationEffect<C extends Component> implements AlterationEffect {
+
+    /**
+     * The component class of the backing component.
+     */
+    protected final Class<C> componentClass;
+
+    private final DelayManager delayManager;
+    private final String effectId;
+
+    /**
+     * Initialize a component-backed alteration effect for the given component class and identifier.
+     *
+     * @param context the game context to retrieve objects without dependency injection
+     * @param componentClass the class of the backing component
+     * @param effectIdentifier the effect identifier used with the delayed expiration event
+     */
+    public ComponentBasedAlterationEffect(Context context, Class<C> componentClass, String effectIdentifier) {
+        this.delayManager = context.get(DelayManager.class);
+        this.componentClass = componentClass;
+        this.effectId = effectIdentifier;
+    }
+
+    /**
+     * Hook method called whenever this effect is applied.
+     * <p>
+     * Upsert denotes the the operation to either insert (i.e., create) or update the backing component. If the
+     * component is already present on the entity the optional value will contain the component. Otherwise, it will be
+     * empty.
+     * <p>
+     * The effect context holds all information passed to {@link AlterationEffect#applyEffect(EntityRef, EntityRef,
+     * String, float, long)}. See {@link EffectContext} for more details.
+     * <p>
+     * Implementations of this method should return the initialized (or updated) component.
+     * <p>
+     * <strong>Note:</strong> Ideally, the corresponding system should act upon an {@link OnEffectModifyEvent} for this
+     * alteration effect. If that is the case, the upsert operation only needs to handle component creation, and setting
+     * the correct state is handled by {@link #updateComponent(OnEffectModifyEvent, Component, EffectContext)}.
```

Comment:
Hm, I'm no longer so sure about this statement 🤔 well, I think it still holds, but it takes away one of the major points of the initial idea of alteration effects - avoid writing a system/event handler to apply an alteration effect... still, the current behavior seems buggy, so I'd like to keep that as a reminder for consumers.